src/ide/code_actions/mod.rs

```diff
@@ -74,13 +77,37 @@ fn get_code_actions_for_diagnostics(
         .flat_map(|(code, diagnostics)| match code.as_str() {
             "E0001" => diagnostics
                 .into_iter()
-                .map(|diagnostic| {
-                    rename_unused_variable::rename_unused_variable(
-                        db,
-                        node,
-                        diagnostic.clone(),
-                        params.text_document.uri.clone(),
-                    )
+                .filter_map(|diagnostic| {
+                    let action = |var_name: &str| {
+                        let action = rename_unused_variable::rename_unused_variable(
+                            var_name,
+                            diagnostic.clone(),
+                            params.text_document.uri.clone(),
+                        );
+
+                        Some(action)
+                    };
+
+                    if_chain!(
+                        if let Some(node) = node.ancestor_of_kind(db, SyntaxKind::PathSegmentSimple);
+                        if let Some(path) = node.parent_of_kind(db, SyntaxKind::ExprPath);
+                        if let Some(_) = path.parent_of_kind(db, SyntaxKind::StatementLet);
+
+                        then {
+                           return action(&path.get_text(db))
+                        }
+                    );
+
+                    if_chain!(
+                        if let Some(ident) = node.ancestor_of_type::<ast::PatternIdentifier>(db);
+                        if let Some(_) = ident.as_syntax_node().parent_of_kind(db, SyntaxKind::StatementLet);
+
+                        then {
+                            return action(&ident.name(db).token(db).text(db))
+                        }
+                    );
+
+                    None
```

Comment:
I think this logic should be part of the action code itself. Otherwise, this `match` will explode in size in the future.